### PR TITLE
feat(module:empty): add empty component

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2711 

There was no `Empty` component. And different components handle empty content separately, via inputs like `nzNotFountContent` or `nzNoResult`.

## What is the new behavior?

- All components which need to render an empty placeholder now use `NzEmbedEmptyComponent` component.
  - `nzNotFoundContent` and other old API would be passed to `NzEmbedEmptyComponent`.
  - there are two ways to set a user default placeholder.
  - components affected: cascader, list, select, table, transter.
  - tree-select doesn't support this now. The interaction should be changed completely. @HsuanXyz .
- Related properties have been removed from i18n files.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

- Due to [this bug](https://github.com/angular/angular/issues/15634), users can't invoke `NzEmptyService`'s method to set a default empty **component**. For that, DI is suggested. 
- i18n should be unified.

### CODE REVIEW

File: components/select/nz-select.module.ts [hunk 1]
```diff
@@ -14,9 +17,32 @@ import { NzSelectUnselectableDirective } from './nz-select-unselectable.directiv
...
 
 @NgModule({
-  imports     : [ CommonModule, FormsModule, OverlayModule, NzI18nModule, NzIconModule],
-  declarations: [ NzOptionPipe, NzSubOptionPipe, NzOptionComponent, NzSelectComponent, NzOptionContainerComponent, NzOptionGroupComponent, NzOptionLiComponent, NzSelectTopControlComponent, NzSelectUnselectableDirective ],
-  exports     : [ NzOptionComponent, NzSelectComponent, NzOptionContainerComponent, NzOptionGroupComponent, NzSelectTopControlComponent ]
+  imports     : [
+    CommonModule,
+    FormsModule,
+    OverlayModule,
+    NzI18nModule,
```
Comment (HsuanXyz) [inline components/select/nz-select.module.ts]:
```diff
- NzI18nModule
```

Comment (wendzhue) [reply components/select/nz-select.module.ts]: Removed

File: components/empty/demo/config.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+order: 2
+title:
+  zh-CN: 全局化配置
+  en-US: Default Config
+---
+
+## zh-CN
+
+自定义图片、描述、附属内容。自定义全局组件的 Empty 样式。
```
Comment (HsuanXyz) [inline components/empty/demo/config.md]:
```diff
- 自定义图片、描述、附属内容
```

File: components/empty/doc/index.en-US.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## API
+
+### nz-empty
+
+| Property | Description | Type | Default |
+| -------- | ----------- | ---- | ------- |
+| `[nzNotFoundImage]` | Customize image. Will tread as image url when string provided | string ｜ TemplateRef | - |
+| `[nzNotFoundContent]` | Custom desciption | string ｜ TemplateRef | - |
+| `[nzNotFoundFooter]` | Custom Footer | string ｜ TemplateRef | - |
```
Comment (HsuanXyz) [inline components/empty/doc/index.en-US.md]: https://github.com/NG-ZORRO/ng-zorro-antd/pull/2734

```md
| `string｜TemplateRef<void>`|
```

File: components/empty/doc/index.en-US.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+### NzEmptyService
+
+| Methods/Properties | Description | Parameters |
+| -------- | ----------- | ---- |
+| `setDefaultEmptyContent` | To set the default empty content | `TemplateRef<void>` ｜ `string` ｜ undefined` |
```
Comment (HsuanXyz) [inline components/empty/doc/index.en-US.md]: 最好不要将 `undefined ` 作为参数，建议在说明里备注 `不传则设为默认`，或者提供一个重置方法更为合理。

File: components/empty/nz-empty.component.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<p class="ant-empty-description">
+  <ng-container *nzStringTemplateOutlet="nzNotFoundContent">
+    {{ nzNotFoundContent || ('Empty.description' | nzI18n) }}
+  </ng-container>
+</p>
+<div class="ant-empty-footer" *ngIf="nzNotFoundFooter">
+  <ng-container *nzStringTemplateOutlet="nzNotFoundFooter">
+    {{ nzNotFoundFooter }}
+  </ng-container>
+  <!--<ng-content></ng-content>-->
```
Comment (HsuanXyz) [inline components/empty/nz-empty.component.html]: 这段没用可以拿掉

File: components/empty/nz-empty.component.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation  : ViewEncapsulation.None,
+  selector       : 'nz-empty',
+  templateUrl    : './nz-empty.component.html',
+  styles         : [ 'nz-empty { display: block; }' ],
+  host           : {
+    '[class.ant-empty]': 'true'
```
Comment (HsuanXyz) [inline components/empty/nz-empty.component.ts]:
```diff
{
- '[class.ant-empty]': 'true'
+ 'class': 'ant-empty'
}
```

File: components/list/nz-list.component.ts [hunk 7]
```diff
@@ -63,6 +59,8 @@ export class NzListComponent implements OnInit, OnChanges, OnDestroy {
 
   @Input() @InputBoolean() nzSplit = true;
 
+  @Input() nzNoResult: string | TemplateRef<void>;
```
Comment (HsuanXyz) [inline components/list/nz-list.component.ts]: 新 API 加到文档里面

File: components/list/nz-list.component.ts [hunk 8]
```diff
@@ -32,10 +29,9 @@ import { NzListGrid } from './interface';
     }
   ` ]
 })
-export class NzListComponent implements OnInit, OnChanges, OnDestroy {
+export class NzListComponent implements OnInit, OnChanges {
   /* tslint:disable-next-line:no-any */
   locale: any = {};
```
Comment (HsuanXyz) [inline components/list/nz-list.component.ts]:
```diff
- locale: any = {};
```

File: components/transfer/nz-transfer.component.ts [hunk 9]
```diff
@@ -155,7 +150,7 @@ export class NzTransferComponent implements OnInit, OnChanges, OnDestroy {
 
   // #endregion
 
-  constructor(private cdr: ChangeDetectorRef, private i18n: NzI18nService) {
+  constructor(private cdr: ChangeDetectorRef) {
```
Comment (HsuanXyz) [inline components/transfer/nz-transfer.component.ts]:
```diff
-  locale: any = {};
```

**Comment by wendzhue** [comment]
@HsuanXyz Please review again.

File: components/components.less [hunk 10]
```diff
@@ -15,6 +15,7 @@
 @import "./divider/style/index.less";
 @import "./drawer/style/index";
 @import "./dropdown/style/index.less";
+@import "./empty/style//index.less";
```
Comment (HsuanXyz) [inline components/components.less]:
```diff
- @import "./empty/style//index.less";
+ @import "./empty/style/index.less";
```

**Comment by wendzhue** [comment]
@HsuanXyz Update.

**Pull request merged**